src/terminal.c

```diff
@@ -3466,14 +3466,11 @@ limit_scrollback(term_T *term, garray_T *gap, int update_buffer)
 	    {
 		curwin = wp;
 		check_cursor();
+		update_topline();
 	    }
 	}
 	curwin = curwin_save;
     }
-
-    // make sure cursor is on a valid line
-    if (curbuf == term->tl_buffer)
-	check_cursor();
```

Comment:
@chrisbra 
I deleted this. I don't think there's a problem, but what do you think?

Reply:
Yeah, I think this might be okay.